integrationtests/src/test/java/io/kroxylicious/proxy/KroxyStandaloneIT.java

```diff
@@ -0,0 +1,128 @@
+            try (var producer = CloseableProducer.wrap(new KafkaProducer<String, String>(Map.of(
+                    ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, proxyAddress,
+                    ProducerConfig.CLIENT_ID_CONFIG, "shouldModifyProduceMessage",
+                    ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class,
+                    ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class,
+                    ProducerConfig.DELIVERY_TIMEOUT_MS_CONFIG, 3_600_000)))) {
+                producer.send(new ProducerRecord<>(TOPIC_1, "my-key", PLAINTEXT)).get();
+                producer.send(new ProducerRecord<>(TOPIC_2, "my-key", PLAINTEXT)).get();
+                producer.flush();
+            }
+
+            ConsumerRecords<String, String> records1;
+            ConsumerRecords<String, String> records2;
+            try (var consumer = CloseableConsumer.wrap(new KafkaConsumer<String, String>(Map.of(
+                    ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, proxyAddress,
+                    ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class,
+                    ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class,
+                    ConsumerConfig.GROUP_ID_CONFIG, "my-group-id",
+                    ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")))) {
+                consumer.subscribe(Set.of(TOPIC_1));
+                records1 = consumer.poll(Duration.ofSeconds(10));
+                consumer.subscribe(Set.of(TOPIC_2));
+                records2 = consumer.poll(Duration.ofSeconds(10));
+            }
+
+            assertEquals(1, records1.count());
+            assertEquals(PLAINTEXT, records1.iterator().next().value());
+            assertEquals(1, records2.count());
+            assertEquals(PLAINTEXT, records2.iterator().next().value());
+        }
+    }
+
+    private static KroxyConfigBuilder baseConfigBuilder(String proxyAddress, String bootstrapServers) {
+        return KroxyConfig.builder()
+                .addToVirtualClusters("demo", new VirtualClusterBuilder()
+                        .withNewTargetCluster()
+                        .withBootstrapServers(bootstrapServers)
+                        .endTargetCluster()
+                        .withNewClusterEndpointConfigProvider()
+                        .withType("StaticCluster")
+                        .withConfig(Map.of("bootstrapAddress", proxyAddress))
+                        .endClusterEndpointConfigProvider()
+                        .build())
+                .addNewFilter().withType("ApiVersions").endFilter()
+                .addNewFilter().withType("BrokerAddress").endFilter();
+    }
+
+    /**
+     * Start proxy as a subprocess, so we have a way to signal it to shut down. If we instead run the
+     * main method in-process, when we terminate it, the backing netty services remain running and
+     * prevent the test from completing.
+     */
```

Comment:
I think this is fine as a one off but if we start needing this as a more generic pattern we should look at managing it through a JUnit extension implementing [CloseableResource](https://junit.org/junit5/docs/5.8.2/api/org.junit.jupiter.api/org/junit/jupiter/api/extension/ExtensionContext.Store.CloseableResource.html)

Reply:
Yep, I imagine this only existing until we come up with some more solid idea of how we distribute kroxy. I think then it would become an external test checking that whatever distribution artifacts we produce work (so exercising the docker image/tarball etc)